Capri.Services/Proposals/ProposalDeleter.cs

```diff
@@ -1,35 +1,69 @@
 ﻿using System;
+using System.Linq;
 using System.Threading.Tasks;
 using Microsoft.EntityFrameworkCore;
+using AutoMapper;
 using Capri.Database;
-using Capri.Database.Entities;
 using Capri.Services.Promoters;
+using Capri.Services.Users;
+using Capri.Web.ViewModels.Proposal;
 
 namespace Capri.Services.Proposals
 {
     public class ProposalDeleter : IProposalDeleter
     {
         private readonly ISqlDbContext _context;
+        private readonly IMapper _mapper;
         private readonly IPromoterGetter _promoterGetter;
+        private readonly IUserGetter _userGetter;
 
-        public ProposalDeleter(ISqlDbContext context, IPromoterGetter promoterGetter)
+        public ProposalDeleter(
+            ISqlDbContext context, 
+            IMapper mapper,
+            IPromoterGetter promoterGetter,
+            IUserGetter userGetter)
         {
             _context = context;
+            _mapper = mapper;
             _promoterGetter = promoterGetter;
+            _userGetter = userGetter;
         }
-        public async Task<IServiceResult<Proposal>> Delete(Guid id)
+
+        public async Task<IServiceResult<ProposalView>> Delete(Guid id)
         {
-            var proposal = await _context.Proposals.FirstOrDefaultAsync(p => p.Id == id);
+            var result = await _userGetter.GetCurrentUser();
+            if(!result.Successful())
+            {
+                var errors = result.GetAggregatedErrors();
+                return ServiceResult<ProposalView>.Error(errors);
+            }
+            
+            var currentUser = result.Body();
+            var promoter = 
+                await _context
+                .Promoters
+                .Include(p => p.Proposals)
+                .FirstOrDefaultAsync(p => p.UserId == currentUser.Id);
```

Comment:
Maybe specify also here that `promoter.Proposals.FirstOrDefault(p => p.Id == id);`

Reply:
done